docs/sequence/PISPTransactionApi.puml

```diff
@@ -71,78 +76,214 @@ rnote right of PISP #Light
+activate Switch
+Switch --> ST8: 202 Accepted
+deactivate ST8
+rnote right of ST8: state: pendingTransactionInitiation
 
-note over Switch
-  Switch kicks off the transaction - DFSP issues ""POST /authorizations""
+note over 3p_DFSP_IBS
+  DFSP handles PISP transaction request
 end note
+Switch -> 3p_DFSP_IBS: POST /thirdpartyRequests/transactions
+activate 3p_DFSP_IBS
+3p_DFSP_IBS --> Switch: 202 Accepted
+rnote right of 3p_DFSP_IBS: state: transactionRequestReceived
+note over 3p_DFSP_IBS
+  **transactionRequestState** is **RECEIVED**
+end note
+deactivate Switch
+3p_DFSP_IBS -> Switch: GET /consents/{consentId}
+activate Switch
+Switch --> 3p_DFSP_IBS: 202 Accepted
+deactivate 3p_DFSP_IBS
+Switch -> AUTHS: GET /consents/{consentId}
+activate AUTHS
+AUTHS --> Switch: 202 Accepted
+deactivate Switch
+AUTHS -> AUTHS: lookup for Consent
+AUTHS -> Switch: PUT /consents/{consentId}
+activate Switch
+Switch --> AUTHS: 202 Accepted
+deactivate AUTHS
+Switch -> 3p_DFSP_IBS: PUT /consents/{consentId}
+activate 3p_DFSP_IBS
+3p_DFSP_IBS --> Switch: 202 Accepted
+deactivate Switch
+3p_DFSP_IBS -> 3p_DFSP_IBS: Lookup **consentId, sourceAccountid and pispId**.\nVerify that they exists, and consent is granted with a valid credential
+3p_DFSP_IBS -> 3p_DFSP_IBS: Generate **transactionId**: '1234-1234-1234-1234'
+note over 3p_DFSP_IBS
+  DFSP informs PISP that **transactionId** has been allocated\n
+end note
+rnote right of 3p_DFSP_IBS #Light
+PUT /thirdpartyRequests/transaction/{id}
+{
+  transactionId: '1234-1234-1234-1234',
+  transactionRequestState: 'RECEIVED'
+}
+end note
+3p_DFSP_IBS -> Switch: PUT /thirdpartyRequests/transaction/{id}
+activate Switch
+Switch --> 3p_DFSP_IBS: 202 Accepted
+Switch -> 3p_PISP_IBS: PUT /thirdpartyRequests/transaction/{id}
+activate 3p_PISP_IBS
+3p_PISP_IBS --> Switch: 202 accepted
+deactivate Switch
+3p_PISP_IBS -> ST8: transactionRequestReceived()
+activate ST8
+rnote right of ST8: state: transactionInitiated
+ST8 -> ST8: store received **transactionId**
+rnote right of ST8: state: pendingTransactionAuthorization
+deactivate ST8
+deactivate 3p_PISP_IBS
 
-Switch -> IBS: ""POST /authorizations""
-IBS -> ST8: ""onPostAuthorizations()""
-IBS --> Switch: ""200 OK""
-rnote right of ST8: ""state: authorizationReceived""
+note over 3p_DFSP_IBS
+  DFSP requires QuotesInformation\n
+end note
+rnote right of 3p_DFSP_IBS: state: pendingQuotesRequest
+3p_DFSP_IBS -> SDK_DFSP_OBS: SYNC POST /quotes
+activate SDK_DFSP_OBS
+SDK_DFSP_OBS -> SDK_DFSP_OBS: retrieve QuotesInformation from Switch
+SDK_DFSP_OBS -> 3p_DFSP_IBS: 200 QuotesInformation
+deactivate SDK_DFSP_OBS
+note over 3p_DFSP_IBS
+  DFSP has the QuotesInformation, they can now ask the PISP for authorization
+  **transactionRequestState** is **PENDING**
+end note
+rnote right of 3p_DFSP_IBS: state: pendingAuthorizationRequest
+3p_DFSP_IBS -> SDK_DFSP_OBS: SYNC POST /authorization
+activate SDK_DFSP_OBS
+SDK_DFSP_OBS -> Switch: POST /authorization
```

Comment:
note: in the latest API draft, `POST /authorizations` is now `POST /thirdpartyRequests/authorizations` - but let's leave this for now, since I don't want to have to touch the transaction-requests-service until we get an e2e transfer working.